File: features/person/src/main/java/com/nivasafinance/features/person/service/PersonWriteService.java
package com.nivasafinance.features.person.service;

import com.nivasafinance.common.dto.AddressData;
import com.nivasafinance.common.dto.AddressRequest;
import com.nivasafinance.common.dto.IdentifierData;
import com.nivasafinance.common.dto.IdentifierRequest;
import com.nivasafinance.features.person.dto.PersonCreateRequest;
import com.nivasafinance.features.person.dto.PersonCreateResponse;
import com.nivasafinance.features.person.dto.PersonUpdateRequest;
import com.nivasafinance.features.person.entity.Person;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public interface PersonWriteService {
    PersonCreateResponse createPerson(PersonCreateRequest personRequest);
    void updatePerson(Long personId, PersonUpdateRequest personUpdateRequest);
    String addAddress(Long personId, AddressRequest request);
    AddressData updateAddress(Long personId, String addressId, AddressRequest request);

    IdentifierData addIdentifier(Long personId, IdentifierRequest request);

    void updateIdentifier(Long personId, UUID identifierId, IdentifierRequest request);

    void deleteIdentifier(Long personId, UUID identifierId);

    void updateCreditBureauFields(Long personId, List<Long> cbEnquiryIds, Person.CreditBureauDetails cbDetails);

    void updateDateOfBirthIfAbsent(Long personId, LocalDate dateOfBirth);

    void deletePerson(String mobileNumber);

    void undoDeletePerson(String mobileNumber);
}

File: features/person/src/main/java/com/nivasafinance/features/person/service/PersonWriteService.kt
package com.nivasafinance.features.person.service

import com.nivasafinance.features.person.dto.EmploymentDetailsCreateRequest
import com.nivasafinance.features.person.dto.EmploymentDetailsResponse
import com.nivasafinance.features.person.dto.EmploymentDetailsUpdateRequest
import com.nivasafinance.features.person.dto.PersonAddressMappingRequest
import com.nivasafinance.features.person.dto.PersonAddressMappingResponse
import com.nivasafinance.features.person.dto.PersonAddressMappingUpdateRequest
import com.nivasafinance.features.person.dto.PersonCreateRequest
import com.nivasafinance.features.person.dto.PersonIdentifierCreateRequest
import com.nivasafinance.features.person.dto.PersonIdentifierResponse
import com.nivasafinance.features.person.dto.PersonIdentifierUpdateRequest
import com.nivasafinance.features.person.dto.PersonUpdateRequest
import java.util.UUID

interface PersonWriteService {

    fun createPerson(request: PersonCreateRequest): UUID
    fun deletePerson(id: UUID)
    fun updatePerson(id: UUID, request: PersonUpdateRequest)

    // Address mapping operations
    fun addAddressToPerson(personId: UUID, request: PersonAddressMappingRequest): PersonAddressMappingResponse
    fun updatePersonAddressMapping(
        personId: UUID,
        addressId: UUID,
        request: PersonAddressMappingUpdateRequest
    ): PersonAddressMappingResponse
    fun removeAddressFromPerson(personId: UUID, addressId: UUID)

    // Identifier operations
    fun createPersonIdentifier(personId: UUID, request: PersonIdentifierCreateRequest): PersonIdentifierResponse
    fun updatePersonIdentifier(id: UUID, request: PersonIdentifierUpdateRequest): PersonIdentifierResponse
    fun deletePersonIdentifier(id: UUID)

    // Employment details operations
    fun createPersonEmploymentDetails(
        personId: UUID,
        request: EmploymentDetailsCreateRequest
    ): EmploymentDetailsResponse
    fun updatePersonEmploymentDetails(
        personId: UUID,
        request: EmploymentDetailsUpdateRequest
    ): EmploymentDetailsResponse
    fun deletePersonEmploymentDetails(personId: UUID)
}
